barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.cpp

```diff
@@ -125,20 +214,43 @@ Witnesses::WitnessStack deserialize_witness_stack(std::vector<uint8_t>&& buf)
             return witness_stack;
         },
         &Witnesses::WitnessStack::bincodeDeserialize);
+    BB_ASSERT_EQ(
+        witness_stack.stack.size(), 1U, "witness_buf_to_witness_vector: expected single WitnessMap in WitnessStack");
+
+    return witness_map_to_witness_vector(witness_stack.stack[0].witness);
 }
 
-// TODO(tom): clean this up.
-uint256_t from_be_bytes(std::vector<uint8_t> const& bytes)
+WitnessVector witness_map_to_witness_vector(Witnesses::WitnessMap const& witness_map)
 {
-    BB_ASSERT_EQ(bytes.size(), 32U, "uint256 constructed from bytes array with invalid length");
-    uint256_t result = 0;
-    for (uint8_t byte : bytes) {
-        result <<= 8;
-        result |= byte;
+    // We need to enforce that:
+    //  - the witness map is in increasing order of witness index
+    //  - all the witness indices are positive
+    bool is_witness_map_increasing = true;
```

Comment:
Is this check buying us anything? I remember you saing Kesha had raised some issue here, can't recall what that was. AFAIK the use of std::map here guarantees the indices are monotonically increasing so I don't think this check can ever fail by definition

Reply:
Maybe his concern was something about a separate set of indices stored elsewhere that purportedly index into this witness_vector potentially indexing beyond it?